### PR TITLE
Adding Tests for Remote Worker and refactoring workers.

### PR DESCRIPTION
**Pull request by pushkalkatara**
This PR resolves the issue - #2678

- [x] Refactor workers remove replicated functions (https://github.com/Cloud-CV/EvalAI/pull/2690/commits/47fcdcca4ced1b4005e84278c84817fd83653540)
* There were functions related to `worker`, `remote_worker` and `rl_worker` which were common and written individually for each script. This introduced replications in code and tests.

- [x] Adding tests for Remote Worker (https://github.com/Cloud-CV/EvalAI/pull/2690/commits/8d5a77788725609631e49f6e6e111ac377b39492)
* Added the following tests for remote worker

This PR refactors the worker-related code, remove the replications.

### CODE REVIEW

**Comment by pushkalkatara** [comment]
Increased code coverage by 0.66% after refactoring. Further adding tests required based on issue #2678

**Comment by pushkalkatara** [comment]
Added tests for `remote_submission_worker`. #2678 solved.

**Comment by pushkalkatara** [comment]
@RishabhJain2018 Please review. 

I noticed several functions being re-written in the three workers. I removed the replications and added them to the worker_utils. Please let me know if this modification is fruitful for the PR or not.

I have added tests for `remote_submission_worker` to solve the issue #2678 . Please review

**Comment by RishabhJain2018** [comment]
Thanks @pushkalkatara for the suggestions and I'll review it soon.

File: scripts/workers/worker_utils.py [hunk 1]
```diff
@@ -1,92 +1,187 @@
...
-            response = requests.request(
-                method=method, url=url, headers=headers, data=data
-            )
-            response.raise_for_status()
-        except requests.exceptions.RequestException:
-            logger.info(
-                "The worker is not able to establish connection with EvalAI"
-            )
-            raise
-        return response.json()
```
Comment (vkartik97) [inline scripts/workers/worker_utils.py]: @pushkalkatara 
Can you please explain this change?

File: scripts/workers/worker_utils.py [hunk 2]
```diff
@@ -1,92 +1,187 @@
...
+    kill_now = False
+
+    def __init__(self):
+        signal.signal(signal.SIGINT, self.exit_gracefully)
+        signal.signal(signal.SIGTERM, self.exit_gracefully)
+
+    def exit_gracefully(self, signum, frame):
+        self.kill_now = True
+
+
```
Comment (vkartik97) [inline scripts/workers/worker_utils.py]: Nice Catch 💯

File: scripts/workers/worker_utils.py [hunk 3]
```diff
@@ -96,3 +191,93 @@ def update_submission_status(self, data, challenge_pk):
...
+def create_dir_as_python_package(directory):
+    """
+        Create a directory and then makes it a python
+        package by creating `__init__.py` file.
+    """
+    create_dir(directory)
+    init_file_path = join(directory, "__init__.py")
+    with open(init_file_path, "w") as init_file:  # noqa
+        # to create empty file
+        pass
```
Comment (vkartik97) [inline scripts/workers/worker_utils.py]: @pushkalkatara I need your suggestion on this: Should you encapsulate this as a class `SubmissionFileHandler` or something on those lines.

CC: @RishabhJain2018

File: scripts/workers/worker_utils.py [hunk 4]
```diff
@@ -1,92 +1,187 @@
...
         self.QUEUE_NAME = QUEUE_NAME
+        self.URLS = {
+            "get_message_from_sqs_queue": "/api/jobs/challenge/queues/{}/",
+            "delete_message_from_sqs_queue": "/api/jobs/queues/{}/",
+            "get_submission_by_pk": "/api/jobs/submission/{}",
+            "get_challenge_phases_by_challenge_pk": "/api/challenges/{}/phases/",
+            "get_challenge_by_queue_name": "/api/challenges/challenge/queues/{}/",
+            "get_challenge_phase_by_pk": "/api/challenges/challenge/{}/challenge_phase/{}",
+            "update_submission_data": "/api/jobs/challenge/{}/update_submission/",
+        }
```
Comment (vkartik97) [inline scripts/workers/worker_utils.py]: Nice change here as well 💯

File: scripts/workers/worker_utils.py (hunk 3 above)
Comment (pushkalkatara) [reply scripts/workers/worker_utils.py]: Yes, that is a good idea. Thanks for the suggestion. I'll update the changes.

File: scripts/workers/worker_utils.py (hunk 1 above)
Comment (pushkalkatara) [reply scripts/workers/worker_utils.py]: Oops! Sorry. I missed this while rearranging between `remote_submission_worker.py` and `rl_submission_worker.py`. Thanks for pointing out. I'll revert back to the one taken from `rl_submission_worker.py`

File: scripts/workers/worker_utils.py (hunk 3 above)
Comment (pushkalkatara) [reply scripts/workers/worker_utils.py]: Maybe we can rename it as `FileHandler` as the utils are being used for challenge and phase data creation as well.

Comment (RishabhJain2018) [reply scripts/workers/worker_utils.py]: Yes, that's a good idea.

Comment (vkartik97) [reply scripts/workers/worker_utils.py]: `FileHandler` is better

File: scripts/workers/worker_utils.py (hunk 1 above)
Comment (pushkalkatara) [reply scripts/workers/worker_utils.py]: Fixed with the recent commit.

File: scripts/workers/worker_utils.py (hunk 3 above)
Comment (pushkalkatara) [reply scripts/workers/worker_utils.py]: Fixed with the recent commit.

**Comment by pushkalkatara** [comment]
Hi, @vkartik97, 
I've made the suggested changes. Please review. 
cc: @RishabhJain2018

**Comment by pushkalkatara** [comment]
Thanks for the review @vkartik97 :D

**Comment by pushkalkatara** [comment]
@vkartik97 I was trying docker submissions but unable to test Docker-based submission. Can you please list down steps on how to make docker submissions. I have created https://github.com/Cloud-CV/evalai-cli/issues/273 issue for the same.

**Pull request closed without merging**